src/cryptoadvance/specter/specter.py

```diff
@@ -135,14 +143,84 @@ def __init__(self, data_folder="./data", config={}):
             "alt_symbol": "BTC",
             "price_provider": "",
             "validate_merkle_proofs": False,
+            "bitcoind": False,
+            "bitcoind_setup": {
+                "stage_progress": -1,
+            },
+            "torbrowser_setup": {
+                "stage_progress": -1,
+            },
         }
 
+        if platform.system() == "Darwin":
+            self.torbrowser_path = os.path.join(
+                self.data_folder, "tor-binaries/tor.real"
+            )
+        elif platform.system() == "Windows":
+            self.torbrowser_path = os.path.join(
+                self.data_folder, "tor-binaries/tor.exe"
+            )
+        else:
+            self.torbrowser_path = os.path.join(self.data_folder, "tor-binaries/tor")
+
+        self.bitcoind_path = os.path.join(
+            self.data_folder, "bitcoin-binaries/bitcoin-0.21.0/bin/bitcoind"
+        )
+
+        if platform.system() == "Windows":
+            self.bitcoind_path += ".exe"
+
         # health check: loads config, tests rpc
         # also loads and checks wallets for all users
         try:
             self.check(check_all=True)
+            rpc_conf = next(
+                (
+                    conf
+                    for conf in detect_rpc_confs(
+                        datadir=os.path.expanduser(self.config["rpc"]["datadir"])
+                    )
+                    if conf["port"] == 8332
+                ),
+                None,
+            )
+            if not rpc_conf:
+                if not self.config["rpc"]["user"]:
+                    self.config["rpc"]["user"] = "bitcoin"
+                    self._save()
+                if not self.config["rpc"]["password"]:
+                    self.config["rpc"]["password"] = secrets.token_urlsafe(16)
+                    self._save()
+                rpc_conf = {
+                    "user": self.config["rpc"]["user"],
+                    "password": self.config["rpc"]["password"],
+                }
+
+            self.bitcoind = BitcoindPlainController(
+                bitcoind_path=self.bitcoind_path,
+                rpcport=8332,
+                network="mainnet",
+                rpcuser=rpc_conf["user"],
+                rpcpassword=rpc_conf["password"],
+            )
+
+            self.tor_daemon = TorDaemonController(
```

Comment:
Yes the reason in both is that so when it gets installed it will already be there. I guess it's not a must, but I just wanted to avoid a case where we call the property and it doesn't exist.

Reply:
Mhhm, i'd prefer that it fails as not failing would be some weird bug, is'nt it? 
Imho, we need to encapsulate more logic into smaller pieces which can be tested separately and make that complexity better manageable.

So what about doing it a bit similiar to the tor_controller property in line 598. If it's not defined, you can raise a more or less thoughtful SpecterError which might even find it's way to the UI ?

Reply:
Ok, makes sense, I'll do it like that